src/main/java/com/questhelper/helpers/quests/perilousmoon/PerilousMoon.java

```diff
@@ -0,0 +1,702 @@
+import net.runelite.api.coords.WorldPoint;
+
+@QuestDescriptor(
+	quest = QuestHelperQuest.PERILOUS_MOON
+)
+public class PerilousMoon extends BasicQuestHelper
+{
+	//Items Required
+	ItemRequirement knife, bigFishingNet, rope, pestleAndMortar, saw, hammer, earthTalisman, waterTalisman,
+		stabWeapon, slashWeapon, crushWeapon;
+
+	ItemRequirement combatGear, staminaPotions, antipoison;
+
+	ItemRequirement buildingSupplies, mossLizardTail, bream, breamScales, moonlightGrub, moonlightGrubPaste,
+		enchantedWaterTalisman, enchantedEarthTalisman, infusedWaterTalisman, infusedEarthTalisman, rawMossLizard;
+
+	Requirement emptyInventory5;
+
+	QuestStep talkToAttala, killSulphurNagua, returnToAttala, enterCamTorum,
+		talkToJessamine, enterNeypotzli;
+
+	QuestStep talkToAttalaInNey, talkToZumaInNey, talkToJessamineInNey;
+
+	QuestStep takeBuildingSupplies, enterPrison, buildPrisonCamp, leavePrison;
+
+	QuestStep enterEarthbound, buildEarthCamp, leaveEarth;
+
+	QuestStep enterStreambound, buildStreamCamp, leaveStream;
+
+	QuestStep talkToJessamineForTalismans, leaveNeypotzli, talkToNahta, talkToBlacksmith,
+		enterNeypotzliAfterBlacksmith, talkToAttalaAfterBlacksmith, useTalismans,
+		useTalismansSidebar, talkToEyatlalli, talkToJessamineAfterEyatlalli;
+
+	QuestStep enterStreamboundAgain, getHerbloreSupplies, collectGrub, useGrubOnPestle, getFishingSupplies, fishBream,
+		useKnifeOnBream, getHunterSupplies, enterSmallEntrance, setTrap, rustleBush, pickUpMossLizard,
+		useKnifeOnLizard, leaveEarthboundWithLizard, talkToEyatlalliWithItems, talkToEyatlalliAfterItems;
+
+	QuestStep enterCamTorumForFights, enterNeypotzliForFights;
+	QuestStep enterEarthboundForFight, enterBlueMoonFight, fightBlueMoon, fightBlueMoonSidebar,
+		enterPrisonForFight, enterBloodMoonFight, fightBloodMoon, fightBloodMoonSidebar,
+		enterStreamboundForFight, enterEclipseMoonFight, fightEclipseMoon, fightEclipseMoonSidebar,
+		talkToJessamineEnd, talkToZumaEnd, finishQuest;
+
+	//Zones
+	Zone camTorum, antechamber, prison, streambound, earthbound, ancientShrine, blueMoonRoom, bloodMoonRoom, eclipseRoom;
+
+	Requirement talkedToBuilders, inCamTorum, inAntechamber, talkedToAttalaInNey, talkedToZumaInNey, inPrison, inStreambound,
+		inEarthbound, madePrisonCamp, madeStreamCamp, madeEarthCamp, inAncientShrine, inNeypotzli, eyatlalliNearby, trapSetup,
+		lizardNearby, usedTail, usedPaste, usedScales, hadTail, hadPaste, hadScales, inBlueMoon, inEclipseMoon,
+		inBloodMoon, defeatedBlueMoon, defeatedBloodMoon, defeatedEclipseMoon;
+
+	@Override
+	public Map<Integer, QuestStep> loadSteps()
+	{
+		Map<Integer, QuestStep> steps = new HashMap<>();
+		setupZones();
+		setupRequirements();
+		setupConditions();
+		setupSteps();
+
+		steps.put(0, talkToAttala);
+
+		steps.put(2, killSulphurNagua);
+
+		steps.put(4, returnToAttala);
+
+		ConditionalStep goTalkToJess = new ConditionalStep(this, enterCamTorum);
+		goTalkToJess.addStep(and(inCamTorum, talkedToBuilders), talkToJessamine);
```

Comment:
I did not need to talk to builders to continue - my varbit value of 9824 stayed at 0 after I spoke with Jessamine
```suggestion
		goTalkToJess.addStep(inCamTorum, talkToJessamine);
```